### ISSUE TITLE
[question]Object size

### ISSUE THREAD

**Issue by Dev-hestabit** [alicevision/Meshroom#1814]
Is there anyway apart from CCtag for scaling 3d model to its real life size?
or how meshroom decide the scale for an object if it depend on image size or focal length ?

**Comment by mugulmd** [comment]
Hey @Dev-hestabit 

Unfortunately no you cannot scale the model to its real life size in Meshroom. 

The scale is decided by the StructureFromMotion in a "random" way (it depends on initialization steps and it may vary a bit during the calculations so we don't have much control over it).

However you can always re-import your 3D model afterwards in a 3D modelling software like Blender to re-scale it 😄

**Comment by mugulmd** [comment]
Do you have any suggestion for how we could "incorporate" a scale information in the reconstruction workflow ?

**Comment by Dev-hestabit** [comment]
Yes, don't know this approach is toward a valid direction or not
https://www.scantips.com/lights/subjectdistance.html
according to this article we can roughly calculate the object size using the camera details so we can calculate the object size using the Image itself and may be resize our model so that the distance between two high dense opposite point is equal to the the object size we measure , it is not perfect but it can roughly resize our model,

**Comment by Dev-hestabit** [comment]
can you or anyone help about adding nodes in command line that are not present in the default pipeline 
like keyframe selection node so we can pass video file also or mesh Denoising to improve final object 
i am not able to use these node in CLI

**Comment by mugulmd** [comment]
> Yes, don't know this approach is toward a valid direction or not https://www.scantips.com/lights/subjectdistance.html according to this article we can roughly calculate the object size using the camera details so we can calculate the object size using the Image itself and may be resize our model so that the distance between two high dense opposite point is equal to the the object size we measure , it is not perfect but it can roughly resize our model,

Sounds interesting, thanks!

**Comment by mugulmd** [comment]
> can you or anyone help about adding nodes in command line that are not present in the default pipeline like keyframe selection node so we can pass video file also or mesh Denoising to improve final object i am not able to use these node in CLI

Yep sure, but since it does not relate to this issue, you should make a new issue for it or search if there's already an issue on the subject. 
I found this one which should contain the answers you're looking for: https://github.com/alicevision/Meshroom/issues/851 
Hope this helps 😄
